pkg/controller/vsphere/actuator.go

```diff
@@ -71,25 +75,34 @@ func (a *Actuator) Create(ctx context.Context, machine *machinev1.Machine) error
 		fmtErr := fmt.Errorf(scopeFailFmt, machine.GetName(), err)
 		return a.handleMachineError(machine, fmtErr, createEventAction)
 	}
-	if err := newReconciler(scope).create(); err != nil {
-		if err := scope.PatchMachine(); err != nil {
-			return err
+
+	// Ensure we're not reconciling a stale machine by checking our task-id.
+	// This is a workaround for a cache race condition.
+	if val, ok := a.TaskIDCache[machine.Name]; ok {
```

Comment:
Did you try running `go test -race` on this? We could have a race here and probably need a lock or `sync.Map`